src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicSafeHandle.cs

```diff
@@ -74,31 +74,20 @@ internal sealed class MsQuicContextSafeHandle : MsQuicSafeHandle
     /// <summary>
     /// Optional parent safe handle, used to increment/decrement reference count with the lifetime of this instance.
     /// </summary>
-    private MsQuicSafeHandle? _parent;
+    private readonly MsQuicSafeHandle? _parent;
 
     public unsafe MsQuicContextSafeHandle(QUIC_HANDLE* handle, GCHandle context, delegate* unmanaged[Cdecl]<QUIC_HANDLE*, void> releaseAction, SafeHandleType safeHandleType, MsQuicSafeHandle? parent = null)
         : base(handle, releaseAction, safeHandleType)
     {
         _context = context;
-        _parent = parent;
-        if (_parent is not null)
+        if (parent is not null)
         {
-            bool release = false;
-            _parent.DangerousAddRef(ref release);
-            if (!release)
-            {
-                if (NetEventSource.Log.IsEnabled())
-                {
-                    NetEventSource.Error(this, $"{this} {_parent} ref count not incremented");
-                }
-                _parent = null;
-            }
-            else
+            bool success = false;
+            parent.DangerousAddRef(ref success);
```

Comment:
Why don't we need to check the `success`? I had it like that originally because I saw the implementation, changed it during one of the PR reviews. What purpose does the `ref success` serve? Is it only backward compat now?